finetrainers/trainer.py

```diff
@@ -660,12 +683,21 @@ def train(self) -> None:
                         text_conditions = batch["text_conditions"]
                         latent_conditions["latents"] = DiagonalGaussianDistribution(
                             latent_conditions["latents"]
-                        ).sample(generator)
-                        if "post_latent_preparation" in self.model_config.keys():
-                            latent_conditions = self.model_config["post_latent_preparation"](**latent_conditions)
-                        align_device_and_dtype(latent_conditions, accelerator.device, weight_dtype)
-                        align_device_and_dtype(text_conditions, accelerator.device, weight_dtype)
-                        batch_size = latent_conditions["latents"].shape[0]
+                        ).sample(torch.Generator(self.state.accelerator.device).manual_seed(self.args.seed))
+                        if is_cog:
+                            if not cog_vae_config["invert_scale_latents"]:
+                                latent_conditions["latents"] = latent_conditions["latents"] * cog_vae_config["scaling_factor"]
+                            else:
+                                latent_conditions["latents"] = 1 / cog_vae_config["scaling_factor"] * latent_conditions["latents"]
+                        
+                    if is_cog:
+                        latent_conditions["latents"] = self._pad_frames(latents=latent_conditions["latents"], denoiser_config=denoiser_config)
```

Comment:
Let me know if this is the correct way to handle the case for Cog 1.5.

Reply:
Should probably go into post_latent_preparation

Reply:
Here it should indeed be appended. It is necessary to confirm whether it is the last frame of the spliced video, not pad 0.

Reply:
Could you provide more details? Perhaps mention the rough changes that might required to the `_pad_frames()` function? I think that would be helpful.